plugin.php

```diff
@@ -286,15 +286,18 @@ function sailthru_user_login( $user_login, $user ) {
 		$client = new WP_Sailthru_Client( $api_key, $api_secret );
 
 		$id      = $user->user_email;
-		$options = array(
-			'login'  => array(
-				'user_agent' => sanitize_text_field( $_SERVER['HTTP_USER_AGENT'] ),
-				'key'        => 'email',
-				'ip'         => sanitize_text_field( $_SERVER['SERVER_ADDR'] ),
-				'site'       => sanitize_text_field ($_SERVER['HTTP_HOST'] ) ,
-			),
-			'fields' => array( 'keys' => 1 ),
-		);
+		if ( isset( $_SERVER['HTTP_USER_AGENT'] ) && isset( $_SERVER['SERVER_ADDR'] ) && isset( $_SERVER['HTTP_HOST'] ) )
```

Comment:
**question**: Should this check for all of them at once? Or should the check be for each individual item?
Ex. individual item on line 293:
```
'user_agent' => isset( $_SERVER['HTTP_USER_AGENT'] ) ? sanitize_text_field( $_SERVER['HTTP_USER_AGENT'] ) : '',
```

My guess is that we don't want to discard all of the login information if we are only missing one of these items from the server.

Reply:
Your suggestion is better.